### PR TITLE
Add extension of documentation

### PR DESCRIPTION
**Pull request by pierre-H**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #1408
| License       | MIT
| Doc PR        | 

Add extended JSON-LD context information in the documentation so we can use schema type [for example](http://schema.org/PropertyValueSpecification) and define some properties.
These information can be used in client generator or other components.

See ticket for more info

### CODE REVIEW

File: src/Hydra/Serializer/DocumentationNormalizer.php [hunk 1]
```diff
@@ -477,6 +477,15 @@ private function getProperty(PropertyMetadata $propertyMetadata, string $propert
         if (null !== $type && !$type->isCollection() && (null !== $className = $type->getClassName()) && $this->resourceClassResolver->isResourceClass($className)) {
             $propertyData['owl:maxCardinality'] = 1;
         }
+        // Add extended JSON-LD context informations
+        // @see https://api-platform.com/docs/core/extending-jsonld-context
+        $jsonldContext = $propertyMetadata->getAttributes()['jsonld_context'] ?? [];
+        foreach ($jsonldContext as $contextKey => $contextValue) {
+            if ($contextKey === '@id' || $contextKey === '@type') {
```
Comment (dunglas) [inline src/Hydra/Serializer/DocumentationNormalizer.php]: Should be yoda style

**Comment by pierre-H** [comment]
@dunglas yes but using another key means that we may need to duplicate definitions in `jsonld_context` and `doc_info` if we want these informations in the documentation and the JSON-LD context.

**Comment by pierre-H** [comment]
@dunglas what should be the name of the field for the doc ?

**Comment by pierre-H** [comment]
I finally use `doc_informations`.

**Comment by Simperfit** [comment]
@dunglas could you please review this ?
@pierre-H could you please rebase ?

File: src/Hydra/Serializer/DocumentationNormalizer.php [hunk 2]
```diff
@@ -478,6 +478,14 @@ private function getProperty(PropertyMetadata $propertyMetadata, string $propert
         if (null !== $type && !$type->isCollection() && (null !== $className = $type->getClassName()) && $this->resourceClassResolver->isResourceClass($className)) {
             $propertyData['owl:maxCardinality'] = 1;
         }
+        // Add doc informations
+        $docInformations = $propertyMetadata->getAttributes()['doc_informations'] ?? [];
+        foreach ($docInformations as $infoKey => $infoValue) {
+            if ('@id' === $infoKey || '@type' === $infoKey) {
```
Comment (dunglas) [inline src/Hydra/Serializer/DocumentationNormalizer.php]: Why excluding those keys (maybe can we just let the user override this if he wants to)?

File: src/Hydra/Serializer/DocumentationNormalizer.php [hunk 3]
```diff
@@ -478,6 +478,14 @@ private function getProperty(PropertyMetadata $propertyMetadata, string $propert
         if (null !== $type && !$type->isCollection() && (null !== $className = $type->getClassName()) && $this->resourceClassResolver->isResourceClass($className)) {
             $propertyData['owl:maxCardinality'] = 1;
         }
+        // Add doc informations
+        $docInformations = $propertyMetadata->getAttributes()['doc_informations'] ?? [];
```
Comment (dunglas) [inline src/Hydra/Serializer/DocumentationNormalizer.php]: I would opt for `docs_extra_keys` (with s to docs for consistency)

**Comment by soyuka** [comment]
Closing as old, lmk if I need to reopen

**Pull request closed without merging**